app/controllers/products_controller.rb

```diff
@@ -3,6 +3,19 @@ def index
         @products = Product.all
         @categories = Category.all
     end
+    def edit
+        @product = Product.find(params[:id])
+    end
+    def update
+        @product = Product.find(params[:id])
+        @product.update_attributes(allowed_params)
+        redirect_to root_path
+    end
+
+    private
+    def allowed_params
```

Comment:
The convention dictates that you call this method `product_params` Not a blocker though.

Reply:
Changed